pkg/skaffold/instrumentation/trace.go

```diff
@@ -91,15 +91,15 @@ func StartTrace(ctx context.Context, name string, attributes ...map[string]strin
 		span.SetAttributes(attribute.Key("source_file").String(fmt.Sprintf("%s:%d", file, ln)))
 		return ctx, span.End
 	}
-	return ctx, func(options ...trace.SpanOption) {}
+	return ctx, func(options ...trace.SpanEndOption) {}
 }
 
 // TraceEndError adds an "error" attribute with value err.Error() to a span during it's end/shutdown callback
 // This fnx is intended to used with the StartTrace callback - "endTrace" when an error occurs during the code path
 // of trace, ex: endTrace(instrumentation.TraceEndError(err)); return nil, err
-func TraceEndError(err error) trace.SpanOption {
+func TraceEndError(err error) trace.SpanEndOption {
 	if traceEnabled {
-		return trace.WithAttributes(attribute.KeyValue{Key: "error", Value: attribute.StringValue(PII(err.Error()))})
+		return trace.WithStackTrace(true)
```

Comment:
now span.End takes SpanEndOption as input, however, we cannot create a SpanEndOption WithAttribute, I'd prefer to leave the code here, we can revisit that when working on exporting metrics to firelog.

Reply:
👍